### ISSUE TITLE
Youtube crashes on Debian

### ISSUE THREAD

**Issue by mcvmcv** [atlas-engineer/nyxt#421]
Hello, I have just tried to install the new .deb package nEXT on my Debian 10 system with stumpwm, and failed. After installing, I had about five unmet dependencies, which I installed with apt. I was then able to run next.

But when I try to run it, I the following message:
`user@computer:~/$ next
 <INFO> [11:01:04] next base.lisp (start) - NEXT:+VERSION+: "1.3.2" 
 <INFO> [11:01:04] next base.lisp (load-lisp-file form-fun-4) -
  Loading configuration from #P"/home/user/.config/next/init.lisp"...
 <INFO> [11:01:04] next remote.lisp (initialize-instance :after remote-interface) -
  D-Bus addresses: (#<Unix socket address: "NIL">)
Unhandled SIMPLE-ERROR in thread #<SB-THREAD:THREAD "Anonymous thread" RUNNING
                                    {1006A1AE53}>:
  No server addresses left to try to open.

Backtrace for: #<SB-THREAD:THREAD "Anonymous thread" RUNNING {1006A1AE53}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<SIMPLE-ERROR "No server addresses left to try to open." {1006A22AC3}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<SIMPLE-ERROR "No server addresses left to try to open." {1006A22AC3}>)
2: (INVOKE-DEBUGGER #<SIMPLE-ERROR "No server addresses left to try to open." {1006A22AC3}>)
3: (ERROR "No server addresses left to try to open.")
4: ((LAMBDA NIL :IN DBUS/PROTOCOLS:OPEN-CONNECTION))
5: (DBUS/CONVENIENCE::CALL-WITH-OPEN-BUS #<CLOSURE (LAMBDA (NEXT::BUS) :IN INITIALIZE-INSTANCE) {1006A0FFCB}> #<event base, 0 FDs monitored, using: #<epoll(4) multiplexer> {1006A0E3A3}> (#<DBUS/TRANSPORT-UNIX:UNIX-SERVER-ADDRESS {1006A0FE83}>))
6: ((LAMBDA NIL :IN INITIALIZE-INSTANCE))
7: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
8: ((FLET SB-UNIX::BODY :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
9: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
10: ((FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
11: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-THREAD::CALL-WITH-MUTEX))
12: (SB-THREAD::CALL-WITH-MUTEX #<CLOSURE (FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE) {7FFFF696ED4B}> #<SB-THREAD:MUTEX "thread result lock" owner: #<SB-THREAD:THREAD "Anonymous thread" RUNNING {1006A1AE53}>> NIL T NIL)
13: (SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE #<SB-THREAD:THREAD "Anonymous thread" RUNNING {1006A1AE53}> NIL #<CLOSURE (LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS) {1006A1ADFB}> NIL)
14: ("foreign function: call_into_lisp")
15: ("foreign function: new_thread_trampoline")

unhandled condition in --disable-debugger mode, quitting`

The /home/user/.config/next directory doesn't exist.

I have tried the two commands from the install page: 
`sudo sysctl -w kernel.unprivileged_userns_clone=1`
`sudo sh -c 'echo 1 > /proc/sys/kernel/unprivileged_userns_clone'`

But I'm a bit of a n00b, so I'm now a bit lost. Any suggestions would be welcome, I'm keen to try nEXT.

**Comment by Ambrevar** [comment]
mcvmcv <notifications@github.com> writes:

> Hello, I have just tried to install the new .deb package nEXT on my
> Debian 10 system with stumpwm, and failed. After installing, I had
> about five unmet dependencies, which I installed with apt.

Can you report which those are?  Thanks!

About your error: can you try running Next this way:

    $ export $(dbus-launch) && next

> I have tried the two commands from the install page: 
> `sudo sysctl -w kernel.unprivileged_userns_clone=1`
> `sudo sh -c 'echo 1 > /proc/sys/kernel/unprivileged_userns_clone'`

No need for this, this is only for the Guix pack, not for the .deb.

**Comment by mcvmcv** [comment]
The unmet dependencies:
```
user@computer:~/Downloads$ sudo dpkg -i next_1.3.2_amd64.deb 
[sudo] password for user: 
Selecting previously unselected package next.
(Reading database ... 338661 files and directories currently installed.)
Preparing to unpack next_1.3.2_amd64.deb ...
Unpacking next (1.3.2) ...
dpkg: dependency problems prevent configuration of next:
 next depends on libwebkit2gtk-4.0-dev (>= 2.24); however:
  Package libwebkit2gtk-4.0-dev is not installed.
 next depends on sqlite; however:
  Package sqlite is not installed.
 next depends on libfixposix-dev; however:
  Package libfixposix-dev is not installed.
 next depends on gir1.2-gst-plugins-base-1.0; however:
  Package gir1.2-gst-plugins-base-1.0 is not installed.
 next depends on xclip; however:
  Package xclip is not installed.

dpkg: error processing package next (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 next
```


```
user@computer:~/Downloads$ sudo apt install libwebkit2gtk-4.0-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 libwebkit2gtk-4.0-dev : Depends: gir1.2-webkit2-4.0 (= 2.24.4-1~deb10u1) but it is not going to be installed
                         Depends: libgtk-3-dev but it is not going to be installed
                         Depends: libsoup2.4-dev (>= 2.40) but it is not going to be installed
                         Depends: libjavascriptcoregtk-4.0-dev (= 2.24.4-1~deb10u1) but it is not going to be installed
 next : Depends: sqlite but it is not going to be installed
        Depends: libfixposix-dev but it is not going to be installed
        Depends: gir1.2-gst-plugins-base-1.0 but it is not going to be installed
        Depends: xclip but it is not going to be installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
user@computer:~/Downloads$ sudo apt --fix-broken install
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following additional packages will be installed:
  freetype2-doc gir1.2-atk-1.0 gir1.2-atspi-2.0 gir1.2-freedesktop gir1.2-gdkpixbuf-2.0 gir1.2-gst-plugins-base-1.0 gir1.2-gstreamer-1.0 gir1.2-gtk-3.0 gir1.2-harfbuzz-0.0 gir1.2-javascriptcoregtk-4.0 gir1.2-pango-1.0 gir1.2-soup-2.4
  gir1.2-webkit2-4.0 libatk-bridge2.0-dev libatk1.0-dev libatspi2.0-dev libblkid-dev libcairo-script-interpreter2 libcairo2-dev libdbus-1-dev libegl1-mesa-dev libepoxy-dev libffi-dev libfixposix-dev libfixposix3 libfontconfig1-dev
  libfreetype6-dev libfribidi-dev libgdk-pixbuf2.0-dev libglib2.0-dev libglib2.0-dev-bin libgraphite2-dev libgtk-3-dev libharfbuzz-dev libharfbuzz-gobject0 libice-dev libjavascriptcoregtk-4.0-dev libmount-dev libpango1.0-dev
  libpixman-1-dev libselinux1-dev libsepol1-dev libsm-dev libsoup2.4-dev libsqlite0 libwayland-bin libwayland-dev libwebkit2gtk-4.0-dev libxcb-shm0-dev libxcomposite-dev libxcursor-dev libxft-dev libxi-dev libxinerama-dev
  libxkbcommon-dev libxml2-dev libxrandr-dev libxrender-dev libxtst-dev pango1.0-tools sqlite sqlite3 uuid-dev wayland-protocols x11proto-composite-dev x11proto-randr-dev x11proto-record-dev x11proto-xinerama-dev xclip
Suggested packages:
  libcairo2-doc libglib2.0-doc libgraphite2-utils libgtk-3-doc libice-doc libpango1.0-doc libsm-doc libsoup2.4-doc libwayland-doc sqlite-doc sqlite3-doc
The following NEW packages will be installed:
  freetype2-doc gir1.2-atk-1.0 gir1.2-atspi-2.0 gir1.2-freedesktop gir1.2-gdkpixbuf-2.0 gir1.2-gst-plugins-base-1.0 gir1.2-gstreamer-1.0 gir1.2-gtk-3.0 gir1.2-harfbuzz-0.0 gir1.2-javascriptcoregtk-4.0 gir1.2-pango-1.0 gir1.2-soup-2.4
  gir1.2-webkit2-4.0 libatk-bridge2.0-dev libatk1.0-dev libatspi2.0-dev libblkid-dev libcairo-script-interpreter2 libcairo2-dev libdbus-1-dev libegl1-mesa-dev libepoxy-dev libffi-dev libfixposix-dev libfixposix3 libfontconfig1-dev
  libfreetype6-dev libfribidi-dev libgdk-pixbuf2.0-dev libglib2.0-dev libglib2.0-dev-bin libgraphite2-dev libgtk-3-dev libharfbuzz-dev libharfbuzz-gobject0 libice-dev libjavascriptcoregtk-4.0-dev libmount-dev libpango1.0-dev
  libpixman-1-dev libselinux1-dev libsepol1-dev libsm-dev libsoup2.4-dev libsqlite0 libwayland-bin libwayland-dev libwebkit2gtk-4.0-dev libxcb-shm0-dev libxcomposite-dev libxcursor-dev libxft-dev libxi-dev libxinerama-dev
  libxkbcommon-dev libxml2-dev libxrandr-dev libxrender-dev libxtst-dev pango1.0-tools sqlite sqlite3 uuid-dev wayland-protocols x11proto-composite-dev x11proto-randr-dev x11proto-record-dev x11proto-xinerama-dev xclip
0 upgraded, 69 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
Need to get 19.6 MB of archives.
After this operation, 72.1 MB of additional disk space will be used.
Do you want to continue? [Y/n] Y
```

```
user@computer:~/Downloads$ sudo apt upgrade
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
The following packages will be upgraded:
  comerr-dev e2fslibs e2fsprogs e2fsprogs-l10n libcom-err2 libcom-err2:i386 libcomerr2 libext2fs2 libss2
9 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/1,741 kB of archives.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] Y
```

**Comment by mcvmcv** [comment]
Your suggested command runs nEXT! I loaded a couple of pages and it seemed to work nicely, but youtube.com caused it to crash.

```
user@computer:~/Downloads$ export $(dbus-launch) && next
 <INFO> [21:22:08] next base.lisp (start) - NEXT:+VERSION+: "1.3.2" 
 <INFO> [21:22:08] next base.lisp (load-lisp-file form-fun-4) -
  Loading configuration from #P"/home/user/.config/next/init.lisp"...
 <INFO> [21:22:08] next remote.lisp (initialize-instance :after remote-interface) -
  D-Bus addresses: (#<Unix socket address: @"/tmp/dbus-yCVRTSG97F">)
 <INFO> [21:22:08] next remote.lisp (initialize-instance :after remote-interface) -
  Bus connection name: :1.1
 <INFO> [21:22:08] next port.lisp (run-program port) -
  Current directory: /home/user/Downloads/
 <INFO> [21:22:08] next port.lisp (run-program port) -
  Platform port path: next-gtk-webkit
 <INFO> [21:22:08] next port.lisp (run-program port) -
  Platform port arguments: NIL
 <INFO> [21:22:08] next port.lisp (run-program port) -
  Platform port log file: /home/user/.local/share/next/next-gtk-webkit.log
 <WARN> [21:22:08] next remote.lisp (current-buffer) - No active window.
 <WARN> [21:22:08] next minibuffer.lisp (%echo-status) -
  Can't echo 'MINIBUFFER-MODE enabled.' without status buffer or interface
 <INFO> [21:22:08] next session.lisp (restore-sexp-session form-fun-4) -
  Restoring (#<BUFFER-DESCRIPTION {100770B833}>
             #<BUFFER-DESCRIPTION {100770BDA3}>
             #<BUFFER-DESCRIPTION {100770C0C3}>)
```

**Comment by Ambrevar** [comment]
What's you graphics card / video driver?

The Youtube crashes seem to be related to Gstreamer / WebKitGTK.
Can you try and see if Epiphany crashes too on Youtube?

You can try to install a couple of gst-plugins* see if that helps.

**Comment by mcvmcv** [comment]
I'm using a GeForce GT 630 Rev. 2. with nouveau. Epiphany seems to run youtube fine.

I've tried installing gstreamer1.0-gtk3 but it doesn't seem to help.

**Comment by vindarel** [comment]
by the way, dpkg doesn't install dependencies, you should have used `apt install ./next.deb` or `gdebi ./next.deb` (we updated the install instructions).

**Comment by Ambrevar** [comment]
This could get fixed in WebKitGTK 2.28.3.
Or with QtWebengine.

**Comment by jmercouris** [comment]
We now have a deb and new versions of WebKitGTk this should no longer be an issue. Please feel free to reopen!
